client/src/hooks/useQuery.ts

```diff
@@ -0,0 +1,50 @@
+import { useCallback } from "react";
+import { useEffect, useState } from "react";
+
+type QueryFunction = () => Promise<any>;
+type QueryOptions = {
+  enabled?: boolean;
+  refetchInterval?: number;
+};
+type QueryState<T> = {
+  status: "idle" | "loading" | "error" | "success";
+  data: T | undefined;
+  error: Error | undefined;
+};
+
+export function useQuery<T>(
+  callback: QueryFunction,
```

Comment:
Why use query is taking callback? 
We want to keep it as minimal as possible.
It should only take, url/method/payload/skip. That's it